src/brevitas/proxy/runtime_quant.py

```diff
@@ -160,21 +168,24 @@ def forward(self, x: Union[Tensor, QuantTensor]) -> Union[Tensor, QuantTensor]:
         elif not self.is_quant_enabled:
             # A tuple helps later with control flows
             # The second None value is used later
-            y = (self.fused_activation_quant_proxy.activation_impl(y), None)
+            y = self.fused_activation_quant_proxy.activation_impl(y)
         else:
             y = self.fused_activation_quant_proxy(y)
-        # If y is an empty IntQuantTensor, we need to check if this is a passthrough proxy,
-        # otherwise return a simple Tensor
 
+        # If y is an empty QuantTensor, we need to check if this is a passthrough proxy,
+        # otherwise return a simple Tensor
         # If the second value (i.e., scale) is None, then quant is disabled
         if isinstance(y, tuple) and y[1] is not None:
             out = self.create_quant_tensor(y)
         elif self.is_passthrough_act and isinstance(x, QuantTensor):
-            # preserve scale/zp/bit/sign even without output quant
-            y = y[0]
+            # preserve quant_metadata
+            if isinstance(y, tuple):
+                y = y[0]
             out = self.create_quant_tensor(y, x=x)
         else:
-            out = y[0]
+            if isinstance(y, tuple):
+                y = y[0]
+            out = y
```

Comment:
I'm struggling with these `tuple` checks. Under what conditions can `y` be an "empty `QuantTensor`"? As far as I can tell, it must be generated by one of the following function calls above:

```python
# Line 166-167
y = self.fused_activation_quant_proxy.activation_impl(y) # Either returns `Tensor` or `QuantTensor`?
y = self.export_handler(y) # returns `Tensor` or `tuple`? Can it be an "empty `QuantTensor`"?
# Line 171
y = self.fused_activation_quant_proxy.activation_impl(y) # returns `Tensor` or `QuantTensor`?
# Line 173
y = self.fused_activation_quant_proxy(y) # returns non-empty `tuple`?
```

If `y` cannot be an empty `tuple`, then these checks should be removed. If `y` can be an empty `QuantTensor`, then I prefer the previous code.

Reply:
Let's define `full` a tuple where there are no None, and `empty` a tuple where only the first value is not None.
In all functions above, the input is always a Tensor, the output is either a Tensor or a Tuple, never a QuantTensor (which is created only later).

```python
y = self.fused_activation_quant_proxy.activation_impl(y)
```
Here, the output is always a Tensor since activation_impl is a normal pytorch function

```python
y = self.export_handler(y)
```
Output is always a full tuple

```python
y = self.fused_activation_quant_proxy(y)
```
Output can be a full or empty tuple, based on whether the quantizer passed to ActQuantProxy is None or not (e.g., QuantReLU(act_quant=None)

So we can have all cases (full tuple, empty tuple, single tensor) when we reach the if/else part